scripts/bug_retriever.py

```diff
@@ -20,161 +20,172 @@ def retrieve_bugs(self, limit: int = None) -> None:
-        timespan_ids = bugzilla.get_ids_between(two_years_and_six_months_ago)
-        if limit:
-            timespan_ids = timespan_ids[-limit:]
-        logger.info(f"Retrieved {len(timespan_ids)} IDs.")
-
-        # Get IDs of labelled bugs.
-        labelled_bug_ids = labels.get_all_bug_ids()
-        if limit:
-            labelled_bug_ids = labelled_bug_ids[-limit:]
-        logger.info(f"{len(labelled_bug_ids)} labelled bugs to download.")
-
-        # Get the commits DB, as we need it to get the bug IDs linked to recent commits.
-        # XXX: Temporarily avoid downloading the commits DB when a limit is set, to avoid the integration test fail when the commits DB is bumped.
-        if limit is None:
-            assert db.download(repository.COMMITS_DB)
-
-        # Get IDs of bugs linked to commits (used for some commit-based models, e.g. backout and regressor).
-        start_date = datetime.now() - relativedelta(years=3)
-        commit_bug_ids = list(
-            set(
-                commit["bug_id"]
-                for commit in repository.get_commits()
-                if commit["bug_id"]
-                and dateutil.parser.parse(commit["pushdate"]) >= start_date
+        try:
+            last_modified = db.last_modified(bugzilla.BUGS_DB)
+        except Exception as e:
+            if str(e) == "Last-Modified is not available":
+                pass
+
+        if last_modified:
+            logger.info(
+                f"Retrieving IDs of bugs modified since the last run on {last_modified}"
             )
-        )
-        if limit:
-            commit_bug_ids = commit_bug_ids[-limit:]
-        logger.info(f"{len(commit_bug_ids)} bugs linked to commits to download.")
-
-        # Get IDs of bugs which are regressions, bugs which caused regressions (useful for the regressor model),
-        # and blocked bugs.
-        regression_related_ids: list[int] = list(
-            set(
-                sum(
-                    (
-                        bug["regressed_by"] + bug["regressions"] + bug["blocks"]
-                        for bug in bugzilla.get_bugs()
-                    ),
-                    [],
+            changed_ids = set(
+                bugzilla.get_ids(
+                    {
+                        "f1": "delta_ts",
+                        "o1": "greaterthaneq",
+                        "v1": last_modified.date(),
+                    }
                 )
             )
-        )
-        if limit:
-            regression_related_ids = regression_related_ids[-limit:]
-        logger.info(
-            f"{len(regression_related_ids)} bugs which caused regressions fixed by commits."
-        )
-
-        # Get IDs of bugs linked to intermittent failures.
-        test_failure_bug_ids = [
-            item["bug_id"]
-            for item in test_scheduling.get_failure_bugs(
-                two_years_and_six_months_ago, datetime.utcnow()
+            logger.info(f"Retrieved {len(changed_ids)} IDs.")
```

Comment:
Only the part related to changed_ids should be behind the `if last_modified`, the rest of the code should be executed no matter if `last_modified` has a value or not

Reply:
Ok

Reply:
Done